src/movies.js

```diff
@@ -1,13 +1,34 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
-
+function getAllDirectors(movies) {
+  const directorsArr = movies.map(function (element) {
+    return element.director;
+  });
+  console.log(directorsArr);
+  return directorsArr;
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(movies) {
+  const spielbergMovies = movies.filter((element) => element.director === 'Steven Spielberg' && element.genre.indexOf ('Drama') > -1);
+  console.log(howManyMovies);
+  return spielbergMovies.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+  function scoresAverage(movies) {
```

Comment:
Just an example:

```
  function scoresAverage(movies) {
    if (movies.length === 0) {
      return 0
    }
    const scores = movies.map(movie => movie.score)
    const sum = scores.reduce((accumalator, currentValue) => {
      if (currentValue && typeof currentValue === 'number'){
        return accumalator + currentValue
      }
      return accumalator
    })
    const average = sum / scores.length
    return parseFloat(average.toFixed(2))
  }
```

Take your time to understand the changes and if needed ask for help